semgrep/semgrep/core_runner.py

```diff
@@ -93,7 +93,7 @@ def get_re_matches(
                     "line": _offset_to_line_no(match.end(), contents),
                     "col": _offset_to_col_no(match.end(), contents),
                 },
-                "extra": {"lines": [contents[match.start() : match.end()]]},
+                "extra": {},
```

Comment:
This was the actual bug. The `lines` information only included the matched portion of the regex, not the whole line. Letting `RuleMatch` grab this information for us fixes the issue.